Divertor monoblock sensitivity (#280)
idaholab/TMAP8: doc/content/examples/divertor_monoblock/sensitivity.md

To accomplish this, sensitivity studies on the [Divertor Monoblock](https://mooseframework.inl.gov/TMAP8/examples/divertor_monoblock/index.html) model described in [!cite](Shimada2024114438) will be performed for (1) steady state conditions based on ITER data [!cite](Hodille2021126003), (2) an inadvertant shutdown transient [!cite](ELMORSHEDY2024101616), and (3) an edge-localized mode transient [!cite](Kessel01092013). The sensitivity studies will be performed using the [MOOSE](https://mooseframework.inl.gov/) Stochastic Module and its Sobol Method according to [!cite](SALTELLI2002280). The results of the sensitivity studies will be evaluated for state limits associated with heat transfer and tritium migration physics included in the [Divertor Monoblock](https://mooseframework.inl.gov/TMAP8/examples/divertor_monoblock/index.html).

> First, the modifications made to the [Divertor Monoblock](https://mooseframework.inl.gov/TMAP8/examples/divertor_monoblock/index.html) model will be described. Second, the conditions the divertor monoblock was subjected to in the steady and transient scenarios will be described. Finally, the results of the sensitivities, namely the first and second order sensitivity indicies and example correlations and state frequencies will be presented.

Reviewer: ```suggestion
First, the modifications made to the [Divertor Monoblock](examples/divertor_monoblock/index.md) model will be described. Second, the conditions the divertor monoblock was subjected to in the steady and transient scenarios will be described. Finally, the results of the sensitivities, namely the first and second order sensitivity indices, example correlations, and state frequencies will be presented.
```

### Divertor Monoblock modifications 

The pulsed operation of the divertor is computationally expensive, so as a first approximation, we replace the pulsed operation with a single steady pulse. We also add several postprocessors.

%Show added postprocessors

### Steady operation 

> % Varied parameters: boundary conditions,

Reviewer: ```suggestion
```

!table id=tab:steady_case

> caption=Steady case varied boundary conditions

Reviewer: ```suggestion
caption=Steady case varied boundary conditions.
```

| Parameter | Samples | Nominal Value | Distribution | Deviation |
| --- | --- | --- | --- | --- |
| Incident Heat Flux | 100 | 10$^{7}$ MW/M$^2$ | Uniform | $\pm$5% |
| Incident Tritium Flux | 100 | 7.90$\times$10$^{-13}$ (normalized) | Uniform | $\pm$5% |
| Coolant Temperature | 100 | 552 K | Uniform | $\pm$5% |
| Coolant Tritium Concentration | 100 | 1.0$\times$10$^{-18}$ (normalized) | Uniform | $\pm$1% |

% parameter space figure

### Transient case: inadvertant shutdown 

!style halign=left

% Transient characteristics (modification to BCs), varied parameters and their ranges

> % Transient figure

Reviewer: doing anything with these comments?

!table id=tab:inadvertant_shutdown_case
caption=Inadvertant shutdown varied parameter space.

| Parameter | Samples | Nominal Value | Distribution | Deviation |
| --- | --- | --- | --- | --- |
| Peak Duration | 1000 | 1s | Normal | $\sigma$=$\pm$5% |
| Peak Heat Flux | 1000 | 2$\times$10$^{7}$ MW/M$^2$ | Normal | $\sigma$=$\pm$5% |
| Coolant Temperature | 1000 | 552 K | Normal | $\sigma$=$\pm$5% |
| Tungsten Conductivity Factor | 1000 | 1.0 | Uniform | $\sigma$=$\pm$5% |

% Parameter space figure

### Transient case: edge-localized mode disruption 

!style halign=left
% Transient characteristics (modification to BCs), varied parameters and their ranges

> % transient figure

Reviewer: address comments

!table id=tab:elm_transient_case
caption=Edge-localized mode transient case parameter space.

| Parameter | Samples | Nominal Value | Distribution | Deviation |
| --- | --- | --- | --- | --- |
| Peak Duration | 1000 | 1.32ms | Normal | $\sigma$=$\pm$5% |
| Peak Heat Flux | 1000 | 1147$\times$10$^{7}$ MW/M$^2$ | Normal | $\sigma$=$\pm$5% |
| Coolant Temperature | 1000 | 552 K | Normal | $\sigma$=$\pm$5% |
| Tungsten Conductivity Factor | 1000 | 1.0 | Uniform | $\sigma$=$\pm$5% |

> % Paramete space figure

Reviewer: remember to add figure

> ## Results 

Reviewer: ```suggestion

## Results 
```

First, the modifications made to the [Divertor Monoblock](examples/divertor_monoblock/index.md) model will be described. Second, the conditions the divertor monoblock was subjected to in the steady and transient scenarios will be described. Finally, the results of the sensitivities, namely the first and second order sensitivity indices, example correlations and state frequencies will be presented.

### Divertor Monoblock modifications 

> The pulsed operation of the divertor is computationally expensive, so as a first approximation, we replace the pulsed operation with a single steady pulse. We also add several postprocessors.

Reviewer: ```suggestion
The pulsed operation of the divertor monoblock is relatively computationally expensive to accurately model since it requires taking small time steps to capture transients. 
For a sensitivity analysis study, where many simulations are required, lowering the cost of individual simulations is paramount. 
So as a first approximation, we replace the pulsed operation with a single steady pulse in this study. 
We also add several postprocessors.
```

Additionally the average temperature in each material

!listing test/tests/divertor_monoblock/steady_state_runner.i block=Postprocessors/avg_temperature_W link=false

!listing test/tests/divertor_monoblock/steady_state_runner.i block=Postprocessors/avg_temperature_Cu link=false

!listing test/tests/divertor_monoblock/steady_state_runner.i block=Postprocessors/avg_temperature_CuCrZr link=false

We also look at the maximum concentration of tritium in each material

!listing test/tests/divertor_monoblock/steady_state_runner.i block=Postprocessors/max_concentration_W link=false

!listing test/tests/divertor_monoblock/steady_state_runner.i block=Postprocessors/max_concentration_Cu link=false

!listing test/tests/divertor_monoblock/steady_state_runner.i block=Postprocessors/max_concentration_CuCrZr link=false

And the total area (2D volume) each material occupies

!listing test/tests/divertor_monoblock/steady_state_runner.i block=Postprocessors/area_W link=false

!listing test/tests/divertor_monoblock/steady_state_runner.i block=Postprocessors/area_Cu link=false

!listing test/tests/divertor_monoblock/steady_state_runner.i block=Postprocessors/area_CuCrZr link=false

> !listing test/tests/divertor_monoblock/steady_state_runner.i block=Postprocessors/total_retention link=false

Reviewer: I suggest adding those to the input file of the main divertor monoblock case. This general pps will be useful across the board for these divertor monoblock simulations.

Reviewer 2: Done

### Steady operation 

The sensitivity studies must sample from a distribution of parameters to determine the net effects of those parameters

> on the model. Here, we vary the incident heat flux, tritium flux, coolant temperature and coolant tritium concentration, sampling from

Reviewer: ```suggestion
on the model. Here, we vary the incident heat flux, tritium flux, coolant temperature, and coolant tritium concentration. For all these parameters, the sampling is done from
```

uniform distributions as shown in [tab:steady_case] and [fig:steady_inputs]. It should be noted that the coolant temperature is unusually high
in this case, as the ITER divertor is designed to have coolant run at 100&deg;C with a 50&deg;C temperature increase at the outlet [!cite](hirai2010iter).

!table id=tab:steady_case caption=Steady case varied boundary conditions.
| Parameter | Samples | Nominal Value | Distribution | Deviation |
| --- | --- | --- | --- | --- |
| Incident Heat Flux | 100 | 10 MW/m$^2$ [!cite](pitts2009iter)| Uniform | $\pm$5% |
| Incident Tritium Flux | 100 | 7.90$\times$10$^{-13}$ (normalized) | Uniform | $\pm$5% |
| Coolant Temperature | 100 | 552 K [!cite](hirai2010iter)| Uniform | $\pm$5% |
| Coolant Tritium Concentration | 100 | 1.0$\times$10$^{-18}$ (normalized) | Uniform | $\pm$1% |

> !media divertor_monoblock_sensitivity.py image_name=divertor_monoblock_sensitivity_figures/steady_state_inputs.png id=fig:steady_inputs 

Reviewer: I suggest updating the labels to more readable names, and adding units. 
Also, the first row is missing numbers for the y axis.

Reviewer 2: Adjusted input file. Advise if more changes are necessary

Reviewer: 1. Limit the number of significant digits on ticks numbers. 
2. Make sure to avoid overlaps between axis title and ticks numbers
3. make points less transparent on off-diagonal plots. They are currently barely visible.

Reviewer: Looks so good now

       style=width:50%;margin-bottom:2%;margin-left:auto;margin-right:auto
       caption=Distribution of sampled steady-state parameter values.

### Transient case: inadvertent shutdown 

!style halign=left

> For a transient case of an inadvertent shutdown, we modify parameters as shown in [tab:inadvertent_shutdown_case] and [fig:shutdown_inputs]. This case is modified such that we assume constant operation for 5.5 hours before encountering a `Peak Heat` (and proportionally scaled tritium) `Flux` for the sampled `Peak Duration` before resuming at one-tenth the heat and tritium flux of the original steady-state.

Reviewer: ```suggestion
For a transient case of an inadvertent shutdown, we modify parameters as shown in [tab:inadvertent_shutdown_case] and [fig:shutdown_inputs]. This case is modified such that we assume constant operation for 5.5 hours before encountering a `Peak Heat Flux` (and proportionally scaled tritium flux) for the sampled `Peak Duration` before resuming at one-tenth the heat and tritium flux of the original steady-state.
```

> !alert! note title=Input file time duration

Reviewer: Maybe justify why this is the case. Is this to limit the length of the tests?

Reviewer: Then, I suggest instead to have the input files set up to run the full amount, but use `cli-args` in the `tests` specification files to limit end_time. 

This way, you can remove this note, and users running these inputs will get the full simulations.

> The input files as contained in the tests directory have an end_time parameter that prevents the entire scenario from being run. This parameter must be changed to some number higher than 2e4+`Peak Duration` in order to model these cases

Reviewer: ```suggestion
The input files as contained in the tests directory have an `end_time` parameter that prevents the entire scenario from being run. This parameter must be changed to some number higher than 2$\times 10^4$ seconds + `Peak Duration` in order to model these cases
```

!alert-end!

The top boundary tritium flux and temperature flux are both adjusted to be a function of time, with functions defined in the input file as follows.

!listing test/tests/divertor_monoblock/shutdown_transient_runner.i block=Functions/mobile_flux_bc_function link=false

!listing test/tests/divertor_monoblock/shutdown_transient_runner.i block=Functions/temperature_flux_bc_function

!table id=tab:inadvertent_shutdown_case caption=Inadvertent shutdown varied parameter space.
| Parameter | Samples | Nominal Value | Distribution | Deviation |
| --- | --- | --- | --- | --- |
| Peak Duration | 1000 | 1s | Normal | $\sigma$=$\pm$5% |
| Peak Heat Flux | 1000 | 20 MW/m$^2$ | Normal | $\sigma$=$\pm$5% |
| Coolant Temperature | 1000 | 552 K | Normal | $\sigma$=$\pm$5% |
| Tungsten Conductivity Factor | 1000 | 0.95 W/m K | Uniform | $\pm$5% |

> !media divertor_monoblock_sensitivity.py image_name=divertor_monoblock_sensitivity_figures/transients_inputs.png id=fig:shutdown_inputs 

Reviewer: Same comment as above about label names and units, as well as the lack of numbers on the top row.

       style=width:50%;margin-bottom:2%;margin-left:auto;margin-right:auto
       caption=Distribution of sampled transient shutdown characteristics.

The results of the simulation are shown in [fig:shutdown_results]. As would be expected, the amount of permeation and the amount of tritium retention have an approximately inverse relationship, and the pairwise interactions across the board are shown. The time at which maximum temperatures occur are all correlated, as one might expect.

> !media divertor_monoblock_sensitivity.py image_name=divertor_monoblock_sensitivity_figures/shutdown_transient_results.png id=fig:shutdown_results style=width:80%;margin-bottom:2%;margin-left:auto;margin-right:auto

Reviewer: Same comment as above about label names and units, as well as the lack of numbers on the top row.

>        caption=Distribution of sampled postprocessor values in the inadvertent shutdown case

Reviewer: I am surprised to see the results here and not in the results section.

### Transient case: edge-localized mode disruption 

!style halign=left

> To simulate an edge-localized mode (ELM) disruption, we simulate a set of input conditions as described in [tab:elm_transient_case]. The heat flux is much larger than the transient shutdown case, but over a shorter duration. Given that the parameters are uncertainties in physical behavior, we model most of these with a normal distribution. The results will be shown below in [fig:elm_correlation]. We use the same tungsten conductivy as in [!cite](Shimada2024114438), but vary the conductivity by a constant factor in the range of $\pm$5%. 

Reviewer: I'm not sure what you mean by `Given that the parameters are uncertainties in physical behavior`, could you please rephrase?

!table id=tab:elm_transient_case caption=Edge-localized mode transient case parameter space.
| Parameter | Samples | Nominal Value | Distribution | Deviation |
| --- | --- | --- | --- | --- |

> | Peak Duration | 1000 | 1.32ms | Normal | $\sigma$=$\pm$5% |

Reviewer: ```suggestion
| Peak Duration | 1000 | 1.32 ms | Normal | $\sigma$=$\pm$5% |
```

| Peak Heat Flux | 1000 | 1147 MW/m$^2$ [!cite](loarte2017elms) | Normal | $\sigma$=$\pm$5% |
| Coolant Temperature | 1000 | 552 K | Normal | $\sigma$=$\pm$5% |
| Tungsten Conductivity Factor | 1000 | 1.0 | Uniform | $\sigma$=$\pm$5% |

## Results 

### Steady operation 

> For purposes of comparison, we look at the maximum observed tungsten temperature and the scaled permeation flux with regards to the heat flux in [fig:ss_correlation]. For the steady-state case, the heat flux and tritium flux are not coupled, though we would expect to see some correlation due to increased diffusivity and trapping considerations. We would also expect increases in heat flux to lead to increased tungsten temperature. While we observe both, the effect of heat flux on tritium permeation rate is more attenuated.

Reviewer: ```suggestion
For purposes of comparison, we look at the maximum observed tungsten temperature and the scaled permeation flux with regards to the heat flux in [fig:ss_correlation]. 
The maximum tungsten temperature naturally correlates with the heat flux, but also depends on thermal conductivity and coolant temperature, as expected.
The link between heat flux and tritium flux is not as strong. In general, larger values of tritium flux to the coolant are obtained with higher heat flux, which is the results from increased diffusivity and temperatures more likely to be sufficient for tritium release from traps. 
However, small tritium flux values are still obtained when a large heat flux is applied. This is attributed to lower temperatures towards the coolant (due to cooler coolant or higher thermal conductivity).
Note, however, that low tritium flux are less likely as the heat flux increases.  
```

!media divertor_monoblock_sensitivity.py image_name=divertor_monoblock_sensitivity_figures/steady_comparison.png id=fig:ss_correlation
       style=width:50%;margin-bottom:2%;margin-left:auto;margin-right:auto
       caption=Correlation betweeen the incident heat flux and the maximum temperature of the tungsten, as well as the total (scaled) permeation flux of tritium into the coolant.

### Inadvertent shutdown 

For the inadvertent shutdown case, we can correlate the input parameters with several output parameters. Perhaps most interesting is the relationship between coolant temperature and the scaled tritium flux, where tritium does not permeate effectively for coolant temperatures below about 550 K. In [fig:shutdown_correlation], the upper plots are scatterplots, the lower are kernel density plots and the diagonal are histograms.

> !media divertor_monoblock_sensitivity.py image_name=divertor_monoblock_sensitivity_figures/shutdown_pairplots.png id=fig:shutdown_correlation

Reviewer: Same comment as above about label names and units, as well as the lack of numbers on the top row.

       style=width:50%;margin-bottom:2%;margin-left:auto;margin-right:auto
       caption=Correlation betweeen the input parameters and several metrics of performance in an inadvertent shutdown scenario.

### Edge-localized mode disruption 

> For the case of simulating an ELM disruption event, we show in [fig:elm_correlation] that though the operating conditions vary, the physics are substantially the same.

Reviewer: ```suggestion
For the case of simulating an ELM disruption event, we show in [fig:elm_correlation] that although the operating conditions vary, the trends are substantially the same.
```

> !media divertor_monoblock_sensitivity.py image_name=divertor_monoblock_sensitivity_figures/elm_pairplots.png id=fig:elm_correlation

Reviewer: Same comment as above about label names and units, as well as the lack of numbers on the top row.

!listing test/tests/divertor_monoblock/steady_state_runner.i block=Postprocessors/avg_temperature_Cu link=false

!listing test/tests/divertor_monoblock/steady_state_runner.i block=Postprocessors/avg_temperature_CuCrZr link=false

We also look at the maximum concentration of tritium in each material

!listing test/tests/divertor_monoblock/steady_state_runner.i block=Postprocessors/max_concentration_W link=false

!listing test/tests/divertor_monoblock/steady_state_runner.i block=Postprocessors/max_concentration_Cu link=false

!listing test/tests/divertor_monoblock/steady_state_runner.i block=Postprocessors/max_concentration_CuCrZr link=false

And the total area (2D volume) each material occupies

!listing test/tests/divertor_monoblock/steady_state_runner.i block=Postprocessors/area_W link=false

!listing test/tests/divertor_monoblock/steady_state_runner.i block=Postprocessors/area_Cu link=false

!listing test/tests/divertor_monoblock/steady_state_runner.i block=Postprocessors/area_CuCrZr link=false

!listing test/tests/divertor_monoblock/steady_state_runner.i block=Postprocessors/total_retention link=false

> 

Reviewer: This needs to be adapted and moved to the original divertor monoblock documentation.

       style=width:50%;margin-bottom:2%;margin-left:auto;margin-right:auto
       caption=Correlation betweeen the input parameters and several metrics of performance in an inadvertent shutdown scenario.

The results of the simulation are shown in [fig:shutdown_results]. As would be expected, the amount of permeation and the amount of tritium retention have an approximately inverse relationship, and the pairwise interactions across the board are shown. The time at which maximum temperatures occur are all correlated, as one might expect.

!media divertor_monoblock_sensitivity.py image_name=divertor_monoblock_sensitivity_figures/shutdown_transient_results.png id=fig:shutdown_results style=width:80%;margin-bottom:2%;margin-left:auto;margin-right:auto
       caption=Distribution of sampled postprocessor values in the inadvertent shutdown case

### Edge-localized mode disruption 

For the case of simulating an ELM disruption event, we show in [fig:elm_correlation] that although the operating conditions vary, the trends are substantially the same.

!media divertor_monoblock_sensitivity.py image_name=divertor_monoblock_sensitivity_figures/elm_pairplots.png id=fig:elm_correlation

>        style=width:50%;margin-bottom:2%;margin-left:auto;margin-right:auto

Reviewer: ```suggestion
       style=width:80%;margin-bottom:2%;margin-left:auto;margin-right:auto
```

>        caption=Correlation betweeen the incident heat flu and the maximum temperature of the tungsten, as well as the total (scaled) permeation flux of tritium t

Reviewer: ```suggestion
       caption=Correlation between the incident heat flux and the maximum temperature of the tungsten, as well as the total (scaled) permeation flux of tritium.
```

The sensitivity studies must sample from a distribution of parameters to determine the net effects of those parameters
on the model. Here, we vary the incident heat flux, tritium flux, coolant temperature and coolant tritium concentration, sampling from
uniform distributions as shown in [tab:steady_case] and [fig:steady_inputs]. It should be noted that the coolant temperature is unusually high

> in this case, as the ITER divertor is designed to have coolant run at 100&deg;C with a 50&deg;C temperature increase at the outlet [!cite](hirai2010iter).

Reviewer: Do you know why such a high value was used, then? Should we use a smaller value instead?

Reviewer 2: I believe that the number that was used is for divertor tritium bake-out with pressurized hot-water flow. Ideally it should be changed, but that should probably be done in a separate issue as it has pretty broad-reaching effects.

Reviewer: Could you add this context to the documentation to justify the current decision?

       style=width:50%;margin-bottom:2%;margin-left:auto;margin-right:auto
       caption=Correlation betweeen the input parameters and several metrics of performance in an inadvertent shutdown scenario.

The results of the simulation are shown in [fig:shutdown_results]. As would be expected, the amount of permeation and the amount of tritium retention have an approximately inverse relationship, and the pairwise interactions across the board are shown. The time at which maximum temperatures occur are all correlated, as one might expect.

!media divertor_monoblock_sensitivity.py image_name=divertor_monoblock_sensitivity_figures/shutdown_transient_results.png id=fig:shutdown_results style=width:80%;margin-bottom:2%;margin-left:auto;margin-right:auto

>        caption=Distribution of sampled postprocessor values in the inadvertent shutdown case

Reviewer: ``` suggestion
       caption=Distribution of sampled postprocessor values in the inadvertent shutdown case.
```
